x-pack/legacy/plugins/code/server/queue/cancellation_service.ts

```diff
@@ -77,9 +81,8 @@ export class CancellationSerivce {
       // 1. Use the cancellation token to pass cancel message to job
       token.cancel();
       // 2. waiting on the actual job promise to be resolved
+      // if jobPromise is rejected, is it safer to just throw the exception and let the caller retry?
```

Comment:
If the `jobPromise` is rejected, we don't need to retry since it's not the cancellation caller's responsibility to retry the actual job. But I think we should have a try-catch block here in case the jobPromise is rejected which is my bad that it's missing now. Otherwise, the caller of this cancellation could be interrupted by the rejected exception here.

Reply:
Added a try-catch block, the exception caught is ignored, as it should be handled by the worker for the job.